frontend/src/components/SocketProvider.js

```diff
@@ -0,0 +1,55 @@
+import React, {
+  createContext, useCallback, useMemo, useState,
+} from 'react';
+import PropTypes from 'prop-types';
+
+export const SocketContext = createContext();
+
+const WS_URL = process.env.REACT_APP_WEBSOCKET_URL;
+
+export const socketPath = (activityReportId, page) => {
+  if (activityReportId && activityReportId !== 'new') {
+    return `/activity-report/edit/${activityReportId}/${page}`;
```

Comment:
Also need to catch the "page" being `undefined`. When a report is first loaded if it doesn't have a page selected you can get a call like `ws://localhost:8080/activity-report/edit/19245/undefined`